### ISSUE TITLE
CompactToken parsing failed issue

### ISSUE THREAD

**Issue by ivanmaria** [microsoftgraph/msgraph-sdk-java#639]
Facing following error "CompactToken parsing failed with error code: 80049217" with status code 401 while trying to hit any graph api using java graph sdk. However, the same api (with same access token) is working with postman. Moreover, This issue has been occurring inconsistently.

### Expected behavior
It should return the result

### Actual behavior
{
  "error": {
    "code": "InvalidAuthenticationToken",
    "message": "CompactToken parsing failed with error code: 80049217",
    "innerError": {
      "date": "2021-02-02T12:53:45",
      "request-id": "97c849e4-f5ce-49d1-a470-7ddbccea7133",
      "client-request-id": "2258e70a-3842-4507-82f1-240af94f3889"
    }
  }
}
### Steps to reproduce the behavior
1. Call an API with java graph sdk
2. Verify the result

 
[AB#7957](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/7957)

**Comment by baywet** [comment]
Hi @ivanmaria 
Thanks for reaching out.
Can you give us more information about the following:
- which version of the SDK are you using?
- which version of the core library are you using?
- which type of account are you using? (office 365 or personal account)
- which library are you using to authenticate?

**Comment by ivanmaria** [comment]
> which version of the SDK are you using?

microsoft-graph-1.9.0

> which version of the core library are you using?

microsoft-graph-core-1.0.0

> which type of account are you using? (office 365 or personal account)

office 365

> which library are you using to authenticate?

msal4j. (The issue also occurs when client credential access token is used)

**Comment by baywet** [comment]
Thanks for all that information. Can you please:
- upgrade to sdk 2.6.0
- upgrade to core 1.0.7
- share the code where you implement the authentication provider with MSAL4J

**Comment by kucera-roman** [comment]
Hello,
we are facing the same issue.
our versions:
- SDK 2.4.1
- core 1.0.5

We are using this API since 19.1.2021 without any problems.
This problem starts on 30.1.2021 and it's random as far as we can tell right now.

We are connecting via ClientCredentialProvider
In Azure portal we registered app which has client secret which never expires.

We can try to upgrade to newer versions as @baywet suggest, but it's strange that the issue occurs only in past few days.

**Comment by baywet** [comment]
@ivanmaria I can see you just closed the issue, does it mean upgrading the SDK solved your problem?

**Comment by ivanmaria** [comment]
@baywet The underlying issue was due to #641. In case of any exceptions, we were retrying to fetch the api. In that process, each time we called the .get() method of any api, the graph sdk would add the authorization header(instead of overwriting it). As a result, multiple authorization header were being added till the third attempt due to which 'CompactToken parsing failed' issue had occurred. However, this error has been observed only since past few days.
